crates/prompt_gateway/src/http_context.rs

```diff
@@ -35,10 +35,10 @@ impl HttpContext for StreamContext {
 
         let request_path = self.get_http_request_header(":path").unwrap_or_default();
         if request_path == HEALTHZ_PATH {
-            if self.embeddings_store.is_none() {
-                self.send_http_response(503, vec![], None);
-            } else {
+            if self.is_embedding_store_initialized() {
```

Comment:
why do we need this

Reply:
this is used in `archgw up` and in e2e tests to make sure we are healthy before users/developers start sending requests to us. See health [check here](https://github.com/katanemo/arch/blob/main/arch/docker-compose.yaml#L17-L21).